deepface/modules/detection.py

```diff
@@ -233,6 +237,17 @@ def detect_faces(
     # find facial areas of given image
     facial_areas = face_detector.detect_faces(img)
 
+    if max_faces is not None and max_faces < len(facial_areas):
+        # sort as largest facial areas come first
+        facial_areas = sorted(
+            facial_areas,
+            key=lambda facial_area: facial_area.w * facial_area.h,
+            reverse=True,
+        )
+        # discard rest of the items
+        facial_areas = facial_areas[0:max_faces]
```

Comment:
I think this code could be more memory and computation efficient by using the heap structure:
```python
from heapq import nlargest

if max_faces is not None and max_faces < len(facial_areas):
    facial_areas = nlargest(
        max_faces,
        facial_areas,
        key=lambda facial_area: facial_area.w * facial_area.h
    )
```
The time complexity of this approach is `O(nlogk)` (where `n` is `len(facial_areas)` and `k` is `max_faces`), compared to sorting, which is `O(nlogn)`. Additionally, the memory consumption here is `O(k)` instead of `O(n)`. 
This optimization could be important when the number of faces is much greater than `max_faces`.

Reply:
OK good idea, changed.